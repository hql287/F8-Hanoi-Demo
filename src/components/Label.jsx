// React Libraries
import React from 'react';
import styled from 'styled-components';
const LabelStyle = styled.p`
  text-transform: uppercase;
  letter-spacing: 1px;
  font-size: 14px;
  font-family: sans-serif;
  font-weight: 600;
  background: black;
  position: relative;
  display: inline-block;
  padding: 4px 10px;
  margin-left: 20px;
  border-radius: 4px;
  color: white;
`;

const Label = ({text}) => <LabelStyle><p>{text}</p></LabelStyle>;

export default Label;
