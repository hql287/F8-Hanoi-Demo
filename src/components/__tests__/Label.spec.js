// Libs
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

// Component to test
import Label from '../Label';

const sampleText = 'Welcome to F8 Hanoi!';
const sampleText2 = 'Hello, World';

describe('Renders correctly to the DOM', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Label text={sampleText} />);
  });

  it('should render one <p> element', () => {
    expect(wrapper.find('p')).toHaveLength(1);
    expect(wrapper.find('p')).not.toHaveLength(2);
  });

  it('should render the components with correct props', () => {
    expect(wrapper.find('p').text()).toEqual(sampleText);
    expect(wrapper.find('p').text()).not.toEqual(sampleText2);
  });

  // Test style Rules
  it('should render component with correct style', () => {

  });

  // Snapshot Testing
  it('matches snapshot', () => {
    const tree = renderer.create(<Label text={sampleText} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
