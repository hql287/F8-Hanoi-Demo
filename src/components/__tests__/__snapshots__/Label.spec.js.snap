// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly to the DOM matches snapshot 1`] = `
<p
  className="Label__LabelStyle-dONxjx iUOqgJ"
>
  <p>
    Welcome to F8 Hanoi!
  </p>
</p>
`;
